File: scala-3/file-share-system/src/main/scala/com/rbleggi/filesharesystem/FileManager.scala
package com.rbleggi.filesharesystem

import scala.collection.mutable

class FileManager {
  private val files = mutable.Map[String, File]()

  def saveFile(file: File): Unit = {
    files(file.name) = file
    println(s"File '${file.name}' saved.")
  }

  def restoreFile(file: File): Unit = {
    if (files.contains(file.name)) {
      println(s"File '${file.name}' restored.")
    } else {
      println(s"File '${file.name}' does not exist.")
    }
  }

  def deleteFile(file: File): Unit = {
    if (files.remove(file.name).isDefined) {
      println(s"File '${file.name}' deleted.")
    } else {
      println(s"File '${file.name}' does not exist.")
    }
  }

  def listFiles(): Unit = {
    if (files.isEmpty) {
      println("\nNo files available.")
    } else {
      println("\nFiles:")
      files.keys.foreach(println)
    }
  }

  def searchFile(query: String): Unit = {
    val results = files.keys.filter(_.contains(query))
    if (results.isEmpty) {
      println(s"\nNo files found matching '$query'.")
    } else {
      println(s"\nFiles matching '$query':")
      results.foreach(println)
    }
  }
}

File: java/file-share/src/main/java/com/rbleggi/filesharesystem/FileManager.java
package com.rbleggi.filesharesystem;

import java.util.HashMap;
import java.util.Map;

public class FileManager {
    private final Map<String, File> files = new HashMap<>();

    public void saveFile(File file) {
        files.put(file.name(), file);
        System.out.println("File '" + file.name() + "' saved.");
    }

    public void restoreFile(File file) {
        if (files.containsKey(file.name())) {
            System.out.println("File '" + file.name() + "' restored.");
        } else {
            System.out.println("File '" + file.name() + "' does not exist.");
        }
    }

    public void deleteFile(File file) {
        if (files.remove(file.name()) != null) {
            System.out.println("File '" + file.name() + "' deleted.");
        } else {
            System.out.println("File '" + file.name() + "' does not exist.");
        }
    }

    public void listFiles() {
        if (files.isEmpty()) {
            System.out.println("\nNo files available.");
        } else {
            System.out.println("\nFiles:");
            files.keySet().forEach(System.out::println);
        }
    }

    public void searchFile(String query) {
        var results = files.keySet().stream()
                .filter(name -> name.contains(query))
                .toList();
        if (results.isEmpty()) {
            System.out.println("\nNo files found matching '" + query + "'.");
        } else {
            System.out.println("\nFiles matching '" + query + "':");
            results.forEach(System.out::println);
        }
    }
}

